Service/Request/RequestFactory.php

```diff
@@ -167,19 +167,31 @@ private function validateEndpointPath(string $path): bool
     }
 
     /**
+     * The query parameters aren't encoded for all GET requests.
+     * For example, "/v1/car-types/built-dates?country=CH&main-type=Qashqai+2&sort=&manufacturer=225" - here 
+     * "Qashqai+2" have to be "Qashqai%2B2".
+     *
      * @param string $path
      *
      * @return array
      */
     private function parseQueryParams(string $path): array
     {
+        $queryParamsArray = [];
         $queryParamsString = parse_url($path, PHP_URL_QUERY);
         if (null !== $queryParamsString) {
+            // $queryParamsArray: ['query-param' => '{queryParam}']
             parse_str($queryParamsString, $queryParamsArray);
 
-            return $queryParamsArray;
+            // $queryParamsArray: ['query-param' => 'queryParam']
+            $queryParamsArray = array_map(function($queryValue) {
```

Comment:
You have to filter out constants, otherwise, they might overlap with the `ServiceRequestInterface`'s fields and be overwritten.

Reply:
good point, thanks